### PR TITLE
Add runtime tests support

### PR DESCRIPTION
**Pull request by Thirumalai-Shaktivel**
Related: #1128

### CODE REVIEW

File: tests/runtime_errors/test_dictionary_01.py [hunk 1]
```diff
@@ -0,0 +1,8 @@
+from ltypes import i32
+
+def test_dictionary():
+    x: dict[i32, str]
+    x = {1: "one", 2: "two"}
+    print(x[10])
```
Comment (Thirumalai-Shaktivel) [inline tests/runtime_errors/test_dictionary_01.py:6]: For the following condition:
https://github.com/lcompilers/lpython/blob/16873d03cfab100232602292a5ba8c78ae0e9b27/src/libasr/codegen/llvm_utils.cpp#L1495-L1522
is this the correct test?
Or 
How should I test KeyError?
@czgdp1807

**Comment by Thirumalai-Shaktivel** [comment]
This is weird, In MacOS the output is captured as *.stderr, but in Ubuntu it is captured as *.stdout

File: tests/runtime_errors/test_dictionary_01.py (hunk 1 above)
Comment (Thirumalai-Shaktivel) [reply tests/runtime_errors/test_dictionary_01.py:6]: But it prints:
```console
$ lpython tests/runtime_errors/test_dictionary_01.py
(null)
```
Is there any other tests?

**Comment by certik** [comment]
This is a bug in LPython. This file:
```
from ltypes import i32

def test_dictionary():
    x: dict[i32, str]
    x = {1: "one", 2: "two"}
    print(x[10])

test_dictionary()
```
gives for me:
```console
$ lpython a.py -o a       
$ ./a
(null)
$ PYTHONPATH=src/runtime/ltypes  python a.py
Traceback (most recent call last):
  File "/Users/ondrej/repos/lpython/a.py", line 8, in <module>
    test_dictionary()
  File "/Users/ondrej/repos/lpython/a.py", line 6, in test_dictionary
    print(x[10])
KeyError: 10
```
So the runtime bounds check does not work in this case. So we have to debug this and fix it.

**Comment by czgdp1807** [comment]
Should we enable this runtime check only in Debug mode only. Should ReleaseSafe also contain it? ReleaseFast shouldn't do this runtime check I think. What do you say?

**Comment by certik** [comment]
Yes, I think ReleaseFast should not have this check, but ReleaseSafe and Debug modes should have it. We can implement this later as well.

**Comment by Thirumalai-Shaktivel** [comment]
I thought *.stderr file will be generated based on the exit_code. But, it doesn't!
> That might be an actual bug that we need to fix in LPython itself.

We have to debug the bug.

**Comment by Thirumalai-Shaktivel** [comment]
The reason can also be that we print the error message to stdout.

**Comment by Thirumalai-Shaktivel** [comment]
Here it is: https://github.com/lcompilers/lpython/blob/4d4b8e682aeca84b371f6526cbdc5e9ba32f08d8/src/libasr/runtime/lfortran_intrinsics.c#L54-L55

**Comment by Thirumalai-Shaktivel** [comment]
@czgdp1807 What do you think, is this the correct fix?

**Comment by Thirumalai-Shaktivel** [comment]
Now, Stderr_hash differs in MacOS!

**Comment by Thirumalai-Shaktivel** [comment]
Hello @certik,
Everything works properly here!
The error output is printed as
```
ld: warning: dylib (/Users/runner/work/lpython/lpython/src/runtime/liblpython_runtime.dylib) was built for newer macOS version (11.7) than being linked (11.0)
ValueError: The list does not contain the element
```
in MacOS
Screenshot from 2022-10-14 19-33-58
This is the reason for the failure!
Why does it prin this warning?

**Comment by certik** [comment]
The macOS failure is due to the extra line in the output:
```
> ld: warning: dylib ($DIR/src/runtime/liblpython_runtime.dylib) was built for newer macOS version (11.7) than being linked (11.0)
```
Which is caused by the `liblpython_runtime.dylib` being compiled for a different mac deployment target than the actual file that LPython itself links (via Clang I think). I think it has to do something with the `MACOSX_DEPLOYMENT_TARGET` variable.

File: .github/workflows/CI.yml [hunk 2]
```diff
@@ -7,7 +7,8 @@ on:
   pull_request:
     branches:
       - main
-
+env:
+  MACOSX_DEPLOYMENT_TARGET: 11.0
```
Comment (Thirumalai-Shaktivel) [inline .github/workflows/CI.yml:11]: This fixed the MacOS CI Failure.

**Review by certik — APPROVED** [review]
This looks good.

**Pull request merged**